chronosphere/tfschema/log_control_config.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Chronosphere Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfschema
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/enum"
+)
+
+var LogControlConfig = map[string]*schema.Schema{
+	"rules": {
+		Type:     schema.TypeList,
+		Elem:     logControlRuleResource,
+		Optional: true,
+	},
+}
+
+var logControlRuleResource = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"mode": Enum{
+			Value:    enum.LogControlRuleMode.ToStrings(),
+			Optional: true,
+		}.Schema(),
+		"filter": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Log query syntax to select logs. Only matching logs will have control action applied.",
```

Comment:
Nit: let's remove the descriptions, since these can easily go out of date without anyone realizing. Better to just rely on the official docs.